main/src/ca/uwaterloo/flix/language/phase/jvm/JvmBackend.scala

```diff
@@ -153,66 +149,9 @@ object JvmBackend {
       effectCallClass,
       effectClasses,
       resumptionWrappers
-    ).reduce(_ ++ _)
-
-    // Write each class (and interface) to disk if enabled.
-    if (flix.options.output.nonEmpty) {
-      for ((_, jvmClass) <- allClasses) {
-        flix.subtask(jvmClass.name.toBinaryName, sample = true)
-        JvmOps.writeClass(flix.options.output.get.resolve("class/"), jvmClass)
-      }
-    }
-
-    // Collect code size for performance tracking.
-    val outputBytes = allClasses.map(_._2.bytecode.length).sum
-
-    if (flix.options.loadClassFiles) {
-      val main = Loader.load(allClasses)
-      new CompilationResult(root, main, getCompiledDefs(root), flix.getTotalTime, outputBytes)
```

Comment:
Its really weird to call `flix.getTotalTime` inside of a call to `flix.phase` - that's why I introduced `Loader.LoaderResult`